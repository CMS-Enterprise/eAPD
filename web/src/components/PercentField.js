import React, { useEffect, useRef } from 'react';
import NumberField from './NumberField';

const PercentField = ({ ...props }) => {
  const ref = useRef(null);
  useEffect(() => {
    if (ref && ref.current) {
      const wrapper = document.createElement('div');
      wrapper.setAttribute('class', 'input-holder__percent');
      ref.current.parentNode.appendChild(wrapper);
      wrapper.appendChild(ref.current);
    }
  });

  return <NumberField {...props} fieldRef={ref} />;
};

export default PercentField;
