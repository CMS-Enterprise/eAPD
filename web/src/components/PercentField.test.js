import { mount } from 'enzyme';
import React from 'react';

import PercentField from './PercentField';

describe('PercentField component', () => {
  it('renders correctly', () => {
    let component;
    component = mount(
      <PercentField
        label="test label"
        name="test name"
        size="medium"
        className="stuff"
        value="123"
        onChange={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
