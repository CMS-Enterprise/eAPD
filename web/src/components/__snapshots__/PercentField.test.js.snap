// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentField component renders correctly 1`] = `
<PercentField
  className="stuff"
  label="test label"
  name="test name"
  onChange={[MockFunction]}
  size="medium"
  value="123"
>
  <NumberField
    className="stuff"
    fieldRef={
      Object {
        "current": <input
          class="ds-c-field ds-c-field--medium"
          id="textfield_1"
          name="test name"
          type="text"
          value="123"
        />,
      }
    }
    label="test label"
    mask={null}
    name="test name"
    onBlur={null}
    onChange={[MockFunction]}
    round={false}
    size="medium"
    value="123"
  >
    <TextField
      className="stuff"
      inputRef={[Function]}
      label="test label"
      mask={null}
      name="test name"
      onBlur={[Function]}
      onChange={[Function]}
      size="medium"
      type="text"
      value="123"
    >
      <div
        className="ds-u-clearfix stuff"
      >
        <FormLabel
          component="label"
          fieldId="textfield_1"
          id="textfield_label_2"
        >
          <label
            className="ds-c-label"
            htmlFor="textfield_1"
            id="textfield_label_2"
          >
            <span
              className=""
            >
              test label
            </span>
          </label>
        </FormLabel>
        <input
          className="ds-c-field ds-c-field--medium"
          id="textfield_1"
          name="test name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="123"
        />
      </div>
    </TextField>
  </NumberField>
</PercentField>
`;
